packages/mongo/mongo_connection.js

```diff
@@ -882,10 +882,11 @@ Object.assign(MongoConnection.prototype, {
           // some newfangled $selector that minimongo doesn't support yet.
           try {
             matcher = new Minimongo.Matcher(cursorDescription.selector);
-            return true;
+            return !!matcher;
           } catch (e) {
-            // XXX make all compilation errors MinimongoError or something
-            //     so that this doesn't ignore unrelated exceptions
+            if (e.message && (e.message.includes('needs an array') )) {
+              throw e;
+            }
```

Comment:
This seems like an unrelated change that I think should be followed up in a separate PR. Also, there's no need to `!!matcher`, as its `constructor` never returns anything, so `new`ing it will always return an instance (i.e., an object).